optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/AbstractSolver.java

```diff
@@ -114,6 +111,54 @@ public void solvingEnded(SolverScope<Solution_> solverScope) {
         phaseLifecycleSupport.fireSolvingEnded(solverScope);
     }
 
+    /**
+     * Used by a phase to notify the solver of its start.
+     * The event should not propagate any further, as that is the job of the phase.
```

Comment:
Confusing. They do propagate  further to termination etc.
I'd just add a comment inside the method at the end, something like:
`// Do not propagate to phases, because the active phase does that for itself and they should not propagate to other phases`